### PR TITLE
Simplification of combinatorial expressions

### PR DESCRIPTION
**Pull request by mohit3011**
Update combsimp.py & tests/tests_combsimp.py

### CODE REVIEW

File: sympy/simplify/tests/test_combsimp.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from sympy import (
-    Rational, combsimp, factorial, gamma, binomial, Symbol, pi, S,
+    Rational, combsimp,factorial, gamma, binomial, Symbol, pi, S,
```
Comment (jksuom) [inline sympy/simplify/tests/test_combsimp.py]: The space before 'factorial' could remain on this line.

File: sympy/simplify/tests/test_combsimp.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
     sin, exp, powsimp, sqrt, sympify, FallingFactorial, RisingFactorial,
     simplify, symbols, cos, rf)
```
Comment (jksuom) [inline sympy/simplify/tests/test_combsimp.py]: `combsimp` can be added to this list.

**Comment by jksuom** [comment]
Tests are failing because `gammasimp` was not imported in the test file.
There are also failing plotting tests, but that is unrelated to this PR.

**Comment by jksuom** [comment]
It seems that `gammasimp` should also be added to [this line](https://github.com/sympy/sympy/blob/master/sympy/simplify/__init__.py#L29).

Then, users of `combsimp` that actually want to simplify gamma-expressions can be directed to `gammasimp` by adding this to the beginning:
```
    if expr.has(gamma):
        return gammasimp(expr)
```
Thereafter, the gamma function need not be considered in `combsimp`. The code can be first simplified by omitting parts having to do with gamma, and then improved to fix the issue at hand.

BTW, the name of this PR could rather be something like 'Simplification of combinatorial expressions'.

**Comment by mohit3011** [comment]
@jksuom , please review the changes.

**Comment by jksuom** [comment]
After this first step, https://github.com/sympy/sympy/pull/12094#issuecomment-275370833 indicates how to proceed. References to `gamma` should be removed from `combsimp`, and the code dealing with purely combinatorial expressions should be added to solve this issue.

**Comment by mohit3011** [comment]
when I'm trying to use 'expr.has' it gives error as expr not defined in test

**Comment by jksuom** [comment]
This is what I get when running bin/test combsimp on your branch:
```
sympy/simplify/tests/test_combsimp.py[4] ....                               [OK]
```
`expr.has` does not seem to give an error.

**Comment by ylemkimon** [comment]
Closing this, since a PR dealing with the same issue #13210 has been merged.

**Pull request closed without merging**